### PR TITLE
HAWQ-755. Fix validateResourcePoolStatus assertion

### PR DESCRIPTION
**Pull request by linwen**
Please review, Thanks!

### CODE REVIEW

**Comment by jiny2** [comment]
+1 LGTM

File: src/backend/resourcemanager/include/resourcepool.h [hunk 1]
```diff
@@ -712,6 +712,10 @@ SimpStringPtr build_segment_status_description(SegStat segstat);
 #define VALIDATE_RATIO_BIAS 	0.005
 #define VALIDATE_RESOURCE_BIAS	0.0001
 
+#define EPSILON 1e-7
+#define IS_DOUBLE_ZERO(d)       (fabs(d) < EPSILON)
+#define IS_DOUBLE_EQ(x, y)      ((fabs((x) - (y))) <= (EPSILON))
```
Comment (huor) [inline src/backend/resourcemanager/include/resourcepool.h]: 1. A tight comparison which defines a "very close with tolerance e" relationship between u and v is:
```
| u - v | <= e * |u| and | u - v | <= e * |v|
```
2. A loose comparison which defines a "close enough with tolerance e" relationship between u and v is:
```
| u - v | <= e * |u| or   | u - v | <= e * |v|
```

**Comment by huor** [comment]
+1

File: src/backend/resourcemanager/include/resourcepool.h (hunk 1 above)
Comment (linwen) [inline src/backend/resourcemanager/include/resourcepool.h]: Thank you!

**Pull request closed without merging**